src/sql/parts/objectExplorer/viewlet/serverTreeActionProvider.ts

```diff
@@ -106,32 +156,51 @@ export class ServerTreeActionProvider extends ContributableActionProvider {
 	/**
 	 * Return actions for OE elements
 	 */
-	public getObjectExplorerNodeActions(tree: ITree, treeNode: TreeNode): IAction[] {
-		let actions = [];
+	private getObjectExplorerNodeActions(context: ObjectExplorerContext): IAction[] {
+		return this.getAllActions(context, (context) => this.getBuiltInNodeActions(context));
+	}
+
+	private getBuiltInNodeActions(context: ObjectExplorerContext): IAction[] {
+		let actions: IAction[] = [];
+		let treeNode = context.treeNode;
 		if (TreeUpdateUtils.isDatabaseNode(treeNode)) {
 			if (TreeUpdateUtils.isAvailableDatabaseNode(treeNode)) {
-				actions.push(this._instantiationService.createInstance(ManageConnectionAction, ManageConnectionAction.ID, ManageConnectionAction.LABEL));
+				actions.push(this._instantiationService.createInstance(ManageConnectionAction, ManageConnectionAction.ID, ManageConnectionAction.LABEL, context.tree));
 			} else {
 				return actions;
 			}
 		}
-		actions.push(this._instantiationService.createInstance(OEAction, NewQueryAction.ID, NewQueryAction.LABEL));
-		let scriptMap: Map<NodeType, any[]> = ObjectExplorerActionUtilities.getScriptMap(treeNode);
-		let supportedActions = scriptMap.get(treeNode.nodeTypeId);
-		let self = this;
-
-		if (supportedActions !== null && supportedActions !== undefined) {
-			supportedActions.forEach(action => {
-				actions.push(self._instantiationService.createInstance(action, action.ID, action.LABEL));
-			});
-		}
-		actions.push(this._instantiationService.createInstance(RefreshAction, RefreshAction.ID, RefreshAction.LABEL, tree, treeNode));
 
+		// TODO: shouldn't New Query only be on Server / Database nodes, not on all nodes in the tree?
```

Comment:
probably better to track this suggestion as a GitHub issue than a code comment

Reply:
Cool - #1890 created. Will remove the comment